src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Weather/Manager.cpp

```diff
@@ -0,0 +1,790 @@
+}
+
+File Manager::getLatestCSSISpaceWeatherFile() const
+{
+    // Parse CSSI Space Weather Directories, e.g.,
+    // `.open-space-toolkit/physics/environment/atmospheric/earth/weather/CSSI-Space-Weather/2022-05-19/`, and find the
+    // latest one.
+
+    Map<Instant, File> CSSISpaceWeatherMap = {};
+
+    for (const auto& directory : this->getCSSISpaceWeatherDirectory().getDirectories())
+    {
+        if ((directory.getName() != temporaryDirectoryName) && directory.containsFileWithName(CSSISpaceWeatherFileName))
+        {
+            const Date date = Date::Parse(directory.getName());
+
+            const Instant instant = Instant::DateTime({date, Time::Midnight()}, Scale::UTC);
+
+            const File bulletinFile = File::Path(directory.getPath() + Path::Parse(CSSISpaceWeatherFileName));
+
+            CSSISpaceWeatherMap.insert({instant, bulletinFile});
+        }
+    }
+
+    if (!CSSISpaceWeatherMap.empty())
+    {
+        return CSSISpaceWeatherMap.rbegin()->second;
+    }
+
+    return const_cast<Manager*>(this)->fetchLatestCSSISpaceWeather_();
+}
+
+void Manager::setup()
+{
+    using ostk::core::fs::Path;
+    using ostk::core::fs::File;
+
+    if (!localRepository_.exists())
+    {
+        localRepository_.create();
+    }
+
+    if (!this->getCSSISpaceWeatherDirectory().exists())
+    {
+        this->getCSSISpaceWeatherDirectory().create();
+    }
+}
+
+void Manager::loadCSSISpaceWeather_(const CSSISpaceWeather& aCSSISpaceWeather)
+{
+    for (const auto& CSSISpaceWeather : CSSISpaceWeatherArray_)
+    {
+        if (CSSISpaceWeather.accessLastObservationDate() == aCSSISpaceWeather.accessLastObservationDate())
+        {
+            throw ostk::core::error::RuntimeError("CSSI Space Weather already added.");
+        }
+    }
+
+    CSSISpaceWeatherArray_.add(aCSSISpaceWeather);  // [TBI] Add in ascending time order
+
+    CSSISpaceWeatherIndex_ = 0;
+}
+
+File Manager::fetchLatestCSSISpaceWeather_()
+{
+    using ostk::core::types::Uint8;
+    using ostk::core::types::Uint16;
+    using ostk::core::types::Integer;
+    using ostk::core::types::String;
+    using ostk::core::ctnr::Map;
+    using ostk::core::fs::Path;
+
+    using ostk::io::ip::tcp::http::Client;
+
+    using ostk::physics::time::Scale;
+    using ostk::physics::time::Date;
+    using ostk::physics::time::Time;
+    using ostk::physics::time::DateTime;
+    using ostk::physics::time::Instant;
+
+    std::cout << "Fetching latest CSSI Space Weather..." << std::endl;
+
+    Directory temporaryDirectory =
+        Directory::Path(this->getCSSISpaceWeatherDirectory().getPath() + Path::Parse(temporaryDirectoryName));
+
+    this->lockLocalRepository(localRepositoryLockTimeout_);
+
+    const URL latestCSSISpaceWeatherUrl = remoteUrl_ + CSSISpaceWeatherFileName;
+    std::cout << latestCSSISpaceWeatherUrl.toString() << std::endl;
+    File latestCSSISpaceWeatherFile = File::Undefined();
+    Directory destinationDirectory = Directory::Undefined();
```

Comment:
```suggestion
    const URL latestCSSISpaceWeatherUrl = remoteUrl_ + CSSISpaceWeatherFileName;
    std::cout << latestCSSISpaceWeatherUrl.toString() << std::endl;

    File latestCSSISpaceWeatherFile = File::Undefined();
    Directory destinationDirectory = Directory::Undefined();
```
Nit: spacing
Also maybe add a header to the std::cout above for some more context?